tool/tool_utils.py

```diff
@@ -0,0 +1,4 @@
+import uuid
+
+def short_id():
+    return str(uuid.uuid4()).split("-")[0]
```

Comment:
Why only include the short id in the audit log?
If we're including a uuid why not the entire uuid4?
I don't mean you should change this... just me learning the reasons behind the decision....

Reply:
Sure

The purpose of this is to enable us to link IDs together using grep, the short IDs are easier to work with and don't have the same impact on making the log unreadable. (c.f. the way in which GitHub's short shas work)

There is other metadata that can be used in the event of a collision - e.g. the name of the machine from which the request originated - so the risk of not being able to determine them is pretty negligible. It may turn out that we need to do machine reading of these logs rather than grep etc, in which case I suggest we enrich the API and use the full UUID